lib/iris/cube.py

```diff
@@ -3055,7 +2983,32 @@ def _order(array):
 
         return cube_xml_element
 
-    def copy(self, data=None):
+    def replace(self, data, dtype=None, fill_value=None):
```

Comment:
@pp-mo I did toy with the idea. We have `Cube.replace_coord`, which suggests `Cube.replace_data` might be appropriate. But I erred on the side that `Cube.replace(data)` reads like "replace my cube data", rather than `Cube.replace_data(data)`, which (to me) reads like `replace data in my cube with data`, which seems like an over specification (in my simple world, and believe me it is simple).

Perhaps, we could argue the toss outside this PR?